ceph-salt-formula/salt/_modules/multi.py

```diff
@@ -0,0 +1,275 @@
+    '''
+    success = []
+    failed = []
+    errored = []
+    slow = []
+    avg = []
+    for result in results:
+        # pylint: disable=invalid-name,unused-variable
+        host, rc, out, err = result
+        if rc == 0:
+            success.append(host)
+            rtt = re.match(
+                r'.*rtt min/avg/max/mdev = \d+\.?\d+/(\d+\.?\d+)/',
+                out, re.DOTALL)
+            if rtt:
+                avg.append({'avg': float(rtt.group(1)), 'host': host})
+        if rc == 1:
+            failed.append(host)
+        if rc == 2:
+            errored.append(host)
+
+    log.debug('multi._summarize_ping average={}'.format(avg))
+
+    if avg:
+        avg_sum = sum(i.get('avg') for i in avg) / len(avg)
+        if len(avg) > 2:
+            for i in avg:
+                if (avg_sum * len(avg) / 2) < i.get('avg'):
+                    log.debug('_summarize_ping: slow host = {} avg = {}, s'.
+                              format(i.get('avg'), avg_sum))
+                    slow.append(i.get('host'))
+    else:
+        avg_sum = 0
+
+    msg = {}
+    msg['succeeded'] = len(success)
+    if failed:
+        msg['failed'] = " ".join(failed)
+    if errored:
+        msg['errored'] = " ".join(errored)
+    if slow:
+        msg['slow'] = " ".join(slow)
+    msg['avg'] = avg_sum
+    return msg
+
+
+def iperf(server, cpu, port):
+    '''
+    iperf test to a specific server
+
+    CLI Example:
+    .. code-block:: bash
+        sudo salt 'node' multi.iperf <hostname>|<ip> <cpu_core> <port>
+    '''
+    log.debug('iperf server ={}'.format(server))
+    return _summarize_iperf(iperf_client_cmd(server, cpu, port))
+
+
+def iperf_client_cmd(server, cpu=0, port=5200):
+    '''
+    Use iperf to test minion to server
+
+    CLI Example:
+    .. code-block:: bash
+    salt 'node' multi.iperf_client_cmd <server_name/ip>
+            cpu=<which_cpu_core default 0> port=<default 5200>
+    '''
+    if IPERF_PATH is None:
+        ret = [LOCALHOST_NAME, 2, "0",
+               "iperf3 not found in path, please install"]
+    elif not server:
+        ret = [LOCALHOST_NAME, 2, "0", "Server name is empty"]
+    else:
+        cmd = ["/usr/bin/iperf3", "-fm", "-A"+str(cpu),
+                     "-t10", "-c"+server, "-p"+str(port)]
+        log.debug('iperf_client_cmd: cmd {}'.format(cmd))
+        proc = Popen(cmd, stdout=PIPE, stderr=PIPE)
+        proc.wait()
+        ret = (server, proc.returncode, proc.stdout.read().decode(), proc.stderr.read().decode())
+    return ret
+
+
+def iperf_server_cmd(cpu=0, port=5200):
+    '''
+    Use iperf to test minion to server
+
+    CLI Example:
+    .. code-block:: bash
+    salt 'node' multi.iperf_server_cmd <server_name/ip>
+        cpu=<which_cpu_core default 0> port=<default 5200>
+    '''
+    if IPERF_PATH is None:
+        return LOCALHOST_NAME + ": iperf3 not found in path, please install"
+    iperf_cmd = ["/usr/bin/iperf3", "-s", "-D", "-A"+str(cpu), "-p"+str(port)]
```

Comment:
Can we do something like `salt-call pkg.install iperf` if iperf is not installed yet?